tests/modules/test_to_control_message.py

```diff
@@ -0,0 +1,164 @@
+
+
+def test_get_module():
+    registry = mrc.ModuleRegistry
+
+    fn_constructor = registry.get_module_constructor("ToControlMessage", "morpheus")
+    assert fn_constructor is not None
+
+    config = {}
+    module_instance = fn_constructor("ToControlMessageTest", config)  # noqa: F841 -- we don't need to use it
+
+
+def test_get_module_with_bad_config():
+
+    def init_wrapper(builder: mrc.Builder):
+
+        def gen_data():
+            for i in range(PACKET_COUNT):
+                yield i
+
+        source = builder.make_source("source", gen_data)
+
+        config = "test"
+        to_control_message = builder.load_module("ToControlMessage", "morpheus", "ToControlMessageTest", config)
+
+        sink = builder.make_sink("sink", on_next, on_error, on_complete)
+
+        builder.make_edge(source, to_control_message.input_port("input"))
+        builder.make_edge(to_control_message.output_port("output"), sink)
+
+    pipeline = mrc.Pipeline()
+    pipeline.make_segment("main", init_wrapper)
+
+    options = mrc.Options()
+    options.topology.user_cpuset = "0-1"
+
+    executor = mrc.Executor(options)
+    executor.register_pipeline(pipeline)
+
+    # This should fail, because no valid module config is specified
+    with pytest.raises(TypeError):
+        executor.start()
+        executor.join()
+
+
+def test_to_control_message_module():
+    packets_received = 0
+    registry = mrc.ModuleRegistry
+    fn_constructor = registry.get_module_constructor("Multiplexer", "morpheus")
+    assert fn_constructor is not None
+
+    def init_wrapper(builder: mrc.Builder):
+
+        df = cudf.DataFrame({
+            'col1': [1, 2, 3, 4, 5],
+            'col2': [1.1, 2.2, 3.3, 4.4, 5.5],
+            'col3': ['a', 'b', 'c', 'd', 'e'],
+            'col4': [True, False, True, False, True]
+        })
+
+        def gen_data():
+            global PACKET_COUNT
+
+            meta = messages.MessageMeta(df)
+
+            for i in range(PACKET_COUNT):
+                yield meta
+
+        def _on_next(control_msg: ControlMessage):
+            nonlocal packets_received
+            packets_received += 1
+            assert (control_msg.payload().df == df)
+
+        source = builder.make_source("source", gen_data)
+
+        config = {"meta_data": {"data_type": "streaming"}, "tasks": [{"type": "inference", "properties": {}}]}
+        to_control_message = builder.load_module("ToControlMessage", "morpheus", "ToControlMessageTest", config)
+
+        sink = builder.make_sink("sink", _on_next, on_error, on_complete)
+
+        builder.make_edge(source, to_control_message.input_port("input"))
+        builder.make_edge(to_control_message.output_port("output"), sink)
+
+    pipeline = mrc.Pipeline()
+    pipeline.make_segment("main", init_wrapper)
+
+    options = mrc.Options()
+
+    options.topology.user_cpuset = "0-1"
+
+    executor = mrc.Executor(options)
+    executor.register_pipeline(pipeline)
+    executor.start()
+    executor.join()
```

Comment:
I dont think we should be using MRC object directly. Refactor this to use Morpheus Pipelines.